cql/src/main/java/org/apache/cassandra/stargate/transport/internal/PlainTextTokenSaslNegotiator.java

```diff
@@ -0,0 +1,113 @@
+package org.apache.cassandra.stargate.transport.internal;
+
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.StoredCredentials;
+import io.stargate.db.AuthenticatedUser;
+import io.stargate.db.Authenticator;
+import io.stargate.db.Persistence;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import org.apache.cassandra.stargate.exceptions.AuthenticationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class PlainTextTokenSaslNegotiator implements Authenticator.SaslNegotiator {
+  private static final Logger logger = LoggerFactory.getLogger(PlainTextTokenSaslNegotiator.class);
+
+  static final String TOKEN_USERNAME = System.getProperty("stargate.cql_token_username", "token");
+  static final int TOKEN_MAX_LENGTH =
+      Integer.parseInt(System.getProperty("stargate.cql_token_max_length", "36"));
+
+  static final byte NUL = 0;
+
+  private final Persistence persistence;
+  private final AuthenticationService authentication;
+  private final Authenticator.SaslNegotiator wrapped;
+  private StoredCredentials credentials;
+  private String username;
+  private String password;
```

Comment:
shouldn't this be a `char[]` for security reasons?

Reply:
Is this related to string duplication? I removed these fields based the suggestion below. `password` will need to be a string at some point though for the `Authentication` API. Maybe we should consider changing that API in another ticket.

Reply:
Yes, it is related to the java string pool (https://www.baeldung.com/java-string-pool). So there is a risk that such a `String password` may leak.

Reply:
+1 on `char[]` but realistically, it will probably become a `String` anyway... either up to down the call chain :)

Reply:
The values are going to be converted to `String` anyway in the case of the wrapped negotiator. I'm not sure this is worth the effort right now. I'm not even sure you can convert a byte array to a char array without using an intermediate `String` anyway.
https://github.com/apache/cassandra/blob/cassandra-3.11.8/src/java/org/apache/cassandra/auth/PasswordAuthenticator.java#L214-L215